core_backend/app/llm_call/parse_input.py

```diff
@@ -0,0 +1,72 @@
+"""
+These are functions that can be used to parse the input questions.
+"""
+
+from ..configs.llm_prompts import (
+    CHECK_INPUT_SAFETY,
+    IDENTIFY_LANG_INPUT,
+    PARAPHRASE_INPUT,
+    TRANSLATE_INPUT,
+    IdentifiedLanguage,
+    SafetyClassification,
+)
+from ..schemas import UserQueryRefined
+from .utils import _ask_llm
+
+
+def input_is_safe(question: UserQueryRefined) -> bool:
+    """
+    Checks for prompt injection in the question.
+    """
+    if (
+        getattr(
+            SafetyClassification,
+            _ask_llm(question.query_text, CHECK_INPUT_SAFETY),
+        )
+        == SafetyClassification.SAFE
+    ):
+        return True
+    return False
+
+
+def identify_language(question: UserQueryRefined) -> UserQueryRefined:
+    """
+    Identifies the language of the question.
+    """
+
+    question.original_language = getattr(
+        IdentifiedLanguage, _ask_llm(question.query_text, IDENTIFY_LANG_INPUT)
+    )
```

Comment:
Would be nice to somehow associate `IdentifiedLanguage` with `IDENTIFY_LANG_INPUT`, and `SafetyClassification` with `CHECK_INPUT_SAFETY`and abstract out the classification. I guess that's what marvin's AI Classifier does but apparently poorly :') So I guess (in sketch)

```
class Classifier:
  prompt: str
  classes: Enum
  def classify(question):
    return getattr(self.classes, _ask_llm(question.query_text, self.prompt)
```

Reply:
Oh interesting idea. Let me see if I can do this quickly in this PR else in a future one